base/compiler/tfuncs.jl

```diff
@@ -1067,6 +1069,20 @@ function _fieldtype_tfunc(@nospecialize(s), exact::Bool, @nospecialize(name))
 end
 add_tfunc(fieldtype, 2, 3, fieldtype_tfunc, 0)
 
+function valid_type_param(T)
```

Comment:
Needs a more specific name to clarify the difference with `valid_tparam`. Maybe `all_instances_valid_tparams`?

Reply:
`valid_tparam_type`?